core/src/main/scala/org/apache/gearpump/cluster/appmaster/ApplicationRuntimeInfo.scala

```diff
@@ -19,20 +19,34 @@
 package org.apache.gearpump.cluster.appmaster
 
 import akka.actor.ActorRef
-import com.typesafe.config.Config
+import com.typesafe.config.{Config, ConfigFactory}
+import org.apache.gearpump.TimeStamp
+import org.apache.gearpump.cluster.ApplicationStatus
 
-import org.apache.gearpump._
-import org.apache.gearpump.cluster.AppMasterRegisterData
-
-/** Run time info used to start an AppMaster */
-case class AppMasterRuntimeInfo(
+/** Run time info of Application */
+case class ApplicationRuntimeInfo(
     appId: Int,
     // AppName is the unique Id for an application
     appName: String,
-    worker: ActorRef = null,
-    user: String = null,
+    appMaster: ActorRef = ActorRef.noSender,
+    worker: ActorRef = ActorRef.noSender,
+    user: String = "",
     submissionTime: TimeStamp = 0,
     startTime: TimeStamp = 0,
     finishTime: TimeStamp = 0,
-    config: Config = null)
-  extends AppMasterRegisterData
+    config: Config = ConfigFactory.empty(),
+    status: ApplicationStatus = ApplicationStatus.Nonexist) {
+
+  def onAppMasterRegister(appMaster: ActorRef, worker: ActorRef): ApplicationRuntimeInfo = {
```

Comment:
Can we align the naming here ? e.g.

* onAppMasterRegistered
* onAppMasterActivated
* onAppMasterTerminated